### PR TITLE
feat(api): membership dashboard api for nestjs

### PR DESCRIPTION
**Pull request by Itrulia**

### CODE REVIEW

File: apps/api-example/src/app/index.ts [hunk 1]
```diff
@@ -292,85 +289,4 @@ export async function runServer(app?: Application | undefined) {
     },
     app
   )
-
```
Comment (Itrulia) [inline apps/api-example/src/app/index.ts:295]: Deleted it so i could turn on strict mode. Was not used anywhere

File: apps/editor/src/app/panel/memberPlanEditPanel.tsx [hunk 2]
```diff
@@ -240,9 +240,9 @@ function MemberPlanEditPanel({id, onClose, onSave}: MemberPlanEditPanelProps) {
       </Drawer.Header>
       <Drawer.Body>
         <Panel>
-          <Form.Group controlId="memberPlanName">
```
Comment (Itrulia) [inline apps/editor/src/app/panel/memberPlanEditPanel.tsx:243]: Fixes membership creation. Already fixed on master but needed it for testing things out

File: apps/editor/src/app/panel/subscriptionEditPanel.tsx [hunk 3]
```diff
@@ -240,7 +241,7 @@ function SubscriptionEditPanel({id, onClose, onSave}: SubscriptionEditPanelProps
     startsAt: startsAt.toISOString(),
     paidUntil: paidUntil ? paidUntil.toISOString() : null,
     properties,
-    deactivation
+    deactivation: deactivation ? stripTypename(deactivation) : deactivation
```
Comment (Itrulia) [inline apps/editor/src/app/panel/subscriptionEditPanel.tsx:244]: Fixes saving a subscription that is deactivated. Currently on master it throws an error

File: apps/editor/tsconfig.json [hunk 4]
```diff
@@ -4,7 +4,6 @@
     "allowJs": false,
     "esModuleInterop": false,
     "allowSyntheticDefaultImports": true,
-    "strict": true,
```
Comment (Itrulia) [inline apps/editor/tsconfig.json:7]: now done in the tsconfig.base.json

**Comment by Itrulia** [comment]
Added you all here so you can comment on the new way to write graphql queries with nestjs.

1 thing still missing from nestjs implementation is authentication @tomaszdurka, something you wanna work on or should I?

File: libs/membership/api/src/lib/dashboard/dashboard.module.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import {Module} from '@nestjs/common'
+import {PrismaModule} from '@wepublish/nest-modules'
+import {DashboardInvoiceResolver} from './invoices/dashboard-invoice.resolver'
+import {DashboardInvoiceService} from './invoices/dashboard-invoice.service'
+import {DashboardSubscriptionResolver} from './subscriptions/dashboard-subscription.resolver'
+import {DashboardSubscriptionService} from './subscriptions/dashboard-subscription.service'
+
```
Comment (michael-scheurer) [inline libs/membership/api/src/lib/dashboard/dashboard.module.ts:7]: The path of this file is endless. Could we simplify? We have now lib in lib folder.

File: libs/membership/api/src/lib/dashboard/invoices/dashboard-invoice.model.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import {Field, Int, ObjectType} from '@nestjs/graphql'
+
```
Comment (michael-scheurer) [inline libs/membership/api/src/lib/dashboard/invoices/dashboard-invoice.model.ts:2]: Even longer path. How about libs/membership/models/dashboard-invoice.model.ts?

File: libs/membership/api/src/lib/dashboard/invoices/dashboard-invoice.service.ts [hunk 7]
```diff
@@ -0,0 +1,102 @@
+import {Injectable} from '@nestjs/common'
+import {PrismaClient} from '@prisma/client'
+import {DashboardInvoice} from './dashboard-invoice.model'
+import {ok} from 'assert'
+
+@Injectable()
+export class DashboardInvoiceService {
```
Comment (michael-scheurer) [inline libs/membership/api/src/lib/dashboard/invoices/dashboard-invoice.service.ts:7]: Very much like, that we use models and services along with classes. But same problem here: Can we simplify the path?

File: apps/editor/src/app/panel/subscriptionEditPanel.tsx (hunk 3 above)
Comment (Itrulia) [reply apps/editor/src/app/panel/subscriptionEditPanel.tsx:244]: Just didn't get around it yet :D

File: libs/membership/api/src/lib/dashboard/dashboard.module.ts (hunk 5 above)
Comment (Itrulia) [reply libs/membership/api/src/lib/dashboard/dashboard.module.ts:7]: Paths shouldnt matter for us. A clear path is better and recommended by nest itself.

File: libs/membership/api/src/lib/dashboard/invoices/dashboard-invoice.model.ts (hunk 6 above)
Comment (Itrulia) [reply libs/membership/api/src/lib/dashboard/invoices/dashboard-invoice.model.ts:2]: That reduces cohesivenees. To change the query you would then need to change multiple directories.

**Review by penk85 — APPROVED** [review]

File: libs/membership/api/setup-database.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import execa from 'execa'
+import {name} from './jest.config'
+
+export default async () => {
+  const databaseUrl = `postgresql://postgres@localhost:5432/${name}?schema=public`
+  process.env.DATABASE_URL = databaseUrl
```
Comment (tomaszdurka) [inline libs/membership/api/setup-database.js:6]: Shouldn't this be take from the process.env ? :o

Comment (Itrulia) [reply libs/membership/api/setup-database.js:6]: This is a test database :D

**Pull request merged**